astropy/utils/decorators.py

```diff
@@ -1101,3 +1100,87 @@ def set_docstring(obj):
         obj.__doc__ = doc.format(*args, **kwargs)
         return obj
     return set_docstring
+
+
+def prepend_docstring(doc, sections=None):
+    """
+    Decorator to prepend the given docstring to the decorated function's
+    after stripping out the list of sections provided.
+
+    Parameters
+    ----------
+    doc : str
+        Docstring to pretend.
+
+    sections : list or None
+        Name of sections to remove from ``doc`` and replace with the
+        decorated doctring one. Default is None to remove no sections.
+    """
+    def decor(func):
+        func.__doc__ = ("\n".join(_remove_sections(doc, sections)) +
+                        textwrap.dedent(func.__doc__))
+        return func
+    return decor
+
+
+def append_docstring(doc, sections=None):
+    """
+    Decorator to append the given docstring to the function's docstr after
+    stripping out the list of sections provided.
+
+    Parameters
+    ----------
+    doc : str
+        Docstring to pretend.
+
+    sections : list or None
+        Name of sections to remove from ``doc`` and replace with the
+        decorated doctring one. Default is None to remove no sections.
+    """
+    def decor(func):
+        func.__doc__ = (textwrap.dedent(func.__doc__) +
+                        "\n".join(_remove_sections(doc, sections)))
+        return func
+    return decor
+
+
+def _remove_sections(doc, sections):
+    """
+    Given a numpydoc-formatted docstring, remove the section blocks provided in
+    ``sections`` and dedent the whole thing.
+
+    Parameters
+    ----------
+    doc : str
+        Docstring to remove sections from.
+
+    sections : list or None
+        Name of sections to remove from ``doc`` and replace with the
+        decorated doctring one. Default is None to remove no sections.
```

Comment:
doctring -> docstring (although I realize that this docstring will never be rendered, but still...)